src/walletConnect/screens/EstimatedNetworkFee.tsx

```diff
@@ -1,69 +1,112 @@
 import * as React from 'react'
 import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
+import SkeletonPlaceholder from 'src/components/SkeletonPlaceholder'
 import TokenDisplay from 'src/components/TokenDisplay'
 import { useSelector } from 'src/redux/hooks'
 import { NETWORK_NAMES } from 'src/shared/conts'
 import Colors from 'src/styles/colors'
 import { typeScale } from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
 import { tokensByIdSelector } from 'src/tokens/selectors'
+import { TokenBalances } from 'src/tokens/slice'
 import { NetworkId } from 'src/transactions/types'
 import Logger from 'src/utils/Logger'
-import { getFeeCurrencyToken, getFeeDecimals, getMaxGasFee } from 'src/viem/prepareTransactions'
+import {
+  getEstimatedGasFee,
+  getFeeCurrencyToken,
+  getFeeDecimals,
+} from 'src/viem/prepareTransactions'
 import {
   SerializableTransactionRequest,
   getPreparedTransaction,
 } from 'src/viem/preparedTransactionSerialization'
 
 interface Props {
+  isLoading: boolean
   networkId: NetworkId
-  transaction: SerializableTransactionRequest
+  transactions: SerializableTransactionRequest[]
 }
 
-export default function EstimatedNetworkFee({ networkId, transaction }: Props) {
+function getNetworkFee(
+  transactions: SerializableTransactionRequest[],
+  networkId: NetworkId,
+  tokensById: TokenBalances
+) {
+  try {
+    const preparedTransactions = transactions.map(getPreparedTransaction)
+    const feeCurrency = getFeeCurrencyToken(preparedTransactions, networkId, tokensById)
+    if (!feeCurrency) {
+      Logger.warn('WalletConnect', 'No fee token info found', { transactions, networkId })
```

Comment:
shall we extract `WalletConnect` to a `TAG` constant? i noticed this pattern in other places.